dluhc-component-library/src/components/maps/DrawingLayer.tsx

```diff
@@ -22,41 +26,67 @@ const DrawingLayer = ({
   strokeWidth = 2,
   circleRadius = 7,
   circleFillColor = "#ffcc33",
+  value,
   onChange,
 }: DrawingLayerProps) => {
   const map = useMap();
-  const source = new VectorSource();
-  const vector = new VectorLayer({
-    source: source,
-    style: {
-      "fill-color": fillcolor,
-      "stroke-color": strokeColor,
-      "stroke-width": strokeWidth,
-      "circle-radius": circleRadius,
-      "circle-fill-color": circleFillColor,
-    },
-  });
-  const ref = useRef<HTMLDivElement>(null);
+
+  const source = useRef(new VectorSource());
+
+  useEffect(() => {
+    const layer = new VectorLayer({
+      source: source.current,
+      style: {
+        "fill-color": fillcolor,
+        "stroke-color": strokeColor,
+        "stroke-width": strokeWidth,
+        "circle-radius": circleRadius,
+        "circle-fill-color": circleFillColor,
+      },
+      zIndex,
+    });
+    map.addLayer(layer);
+    return () => map?.removeLayer(layer);
+  }, [
+    map,
+    source,
+    fillcolor,
+    strokeColor,
+    strokeWidth,
+    circleRadius,
+    circleFillColor,
+    zIndex,
+  ]);
 
   useEffect(() => {
-    if (!ref.current) {
-      return;
+    source.current.clear(); // for now, only allow 1 boundary to be drawn
+    if (value) {
+      source.current.addFeature(
+        new Feature({ geometry: new Polygon(value as number[][][]) }),
```

Comment:
Should this be `as Polygon` ?

Reply:
It accepts `Coordinate[][]` which is just `number[][][]` but doesn't like the `ReadonlyArrays` so I've just cast it to this instead of having to map/slice the deeply nested arrays.